client.go

```diff
@@ -553,6 +573,8 @@ func (c *Client) Stop() {
 		return
 	}
 
+	atomic.StoreInt32(&c.wantStop, 1)
```

Comment:
Sounds reasonable, will place it first in `Stop()`

Reply:
Would that make any difference? Since `runForever` sets it to `0`? So calling `Stop()` before sending a request would just cause the client to start like normal.

Reply:
No the end result wont differ but if you were to inspect the value of `wantStop` it would always be the same before vs. after calling `runForever`. I chatted with @sieuwerts and like I told him I initially thought about calling the variabel `hasCalledStop` but then I thought that didn't matter.

But like my comment reads it's a question about if the variable should symbol if someone has called `Stop()` and if so it should always be the same after calling `Stop()`. If it's not to symbol that, feel free to put it wherever.